Polyfills/Console/Source/Console.cpp

```diff
@@ -1,100 +1,106 @@
-    }
-
-    Console::Console(const Napi::CallbackInfo& info)
-        : ParentT{info}
-        , m_callback{}
-    {
-    }
-
-    void Console::Log(const Napi::CallbackInfo& info)
-    {
-        InvokeEngineCallback("log", info);
-        InvokeCallback(info, Babylon::Polyfills::Console::LogLevel::Log);
-    }
+            for (size_t i = 0; i < argc; ++i)
+            {
+                staticArgs[i] = info[i];
+            }
 
-    void Console::Warn(const Napi::CallbackInfo& info)
-    {
-        InvokeEngineCallback("warn", info);
-        InvokeCallback(info, Babylon::Polyfills::Console::LogLevel::Warn);
-    }
+            func.Call(argc, staticArgs.data());
+        }
+        else
+        {
+            std::vector<Napi::Value> args(argc);
+            for (size_t i = 0; i < argc; ++i)
+            {
+                args[i] = info[i];
+            }
 
-    void Console::Error(const Napi::CallbackInfo& info)
-    {
-        InvokeEngineCallback("error", info);
-        InvokeCallback(info, Babylon::Polyfills::Console::LogLevel::Error);
+            func.Call(argc, args.data());
+        }
     }
 
-    void Console::InvokeCallback(const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel) const
+    void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
         std::stringstream ss{};
-        for (unsigned int index = 0; index < info.Length(); index++)
+        for (size_t i = 0; i < info.Length(); i++)
         {
-            if (index > 0)
+            if (i > 0)
             {
                 ss << " ";
             }
-            ss << info[index].ToString().Utf8Value().c_str();
+            ss << info[i].ToString().Utf8Value().c_str();
         }
         ss << std::endl;
-        m_callback(ss.str().c_str(), logLevel);
-    }
 
-    void Console::InvokeEngineCallback(const std::string functionName, const Napi::CallbackInfo& info)
-    {
-        if (m_engineConsole != nullptr)
-        {
-            const auto engineConsoleFunc = m_engineConsole->Value().Get(functionName).As<Napi::Function>();
-            if (!engineConsoleFunc.IsUndefined()) {
-                const auto args = GetCallbackInfoArgs(info);
-                engineConsoleFunc.As<Napi::Function>().Call(m_engineConsole->Value(), args.size(), args.data());
-            }
-        }
+        callback(ss.str().c_str(), logLevel);
     }
 }
 
 namespace Babylon::Polyfills::Console
 {
     void Initialize(Napi::Env env, CallbackT callback)
     {
-        Internal::Console::CreateInstance(env, std::move(callback));
+        Napi::HandleScope scope{env};
+
+        auto console = env.Global().Get(JS_INSTANCE_NAME).As<Napi::Object>();
+        if (console.IsUndefined())
+        {
+            console = Napi::Object::New(env);
+
+            console.Set("log", Napi::Function::New(env, [callback](const Napi::CallbackInfo& info)
```

Comment:
I would refactor these calls to console.set inside a new function to avoid copy/paste and improve readability.